### PR TITLE
Make sure we save attachment file extension

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by rowanseymour** [comment]
@norkans7 so this is fixing the IVR attachments with broken links issue? What exactly is happening there? Is Twilio giving us media files without extensions or without content-type in the HTTP response?

**Comment by norkans7** [comment]
Yes, the media URLs are without extension.

**Comment by rowanseymour** [comment]
From https://www.twilio.com/docs/voice/twiml/record

 > A request to the RecordingUrl will return a recording in binary WAV audio format by default. To request the recording in MP3 format, append ".mp3" to the RecordingUrl.

Wouldn't it be easier to just add .mp3 to URL, then save an attachment with `audio/mp3` content type?

**Comment by norkans7** [comment]
I think we can do that in the IVR case probably, this fix is the same approach we use on courier when do not know the extension, adding to always request mp3 for IVR for twilio is fine too.

**Comment by rowanseymour** [comment]
Mailroom only handles attachments from Twilio, Nexmo and the ticketing services.. so if they give us explicit content-types those seem preferable to sniffing. If we find a case where that doesn't work we can add sniffing.

File: core/ivr/twiml/twiml.go [hunk 1]
```diff
@@ -251,7 +251,7 @@ func (c *client) InputForRequest(r *http.Request) (string, utils.Attachment, err
 		if url == "" {
 			return "", ivr.NilAttachment, nil
 		}
-		return "", utils.Attachment("audio:" + r.Form.Get("RecordingUrl")), nil
+		return "", utils.Attachment("audio:" + url + ".mp3"), nil
```
Comment (rowanseymour) [inline core/ivr/twiml/twiml.go]: audio/mp3:

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Comment by nicpottier** [comment]
So was this a change on the Twilio side?

**Comment by norkans7** [comment]
I think it has always been so

**Comment by nicpottier** [comment]
So what caused the new issue from Tim noticed?

**Comment by norkans7** [comment]
They use Zapier and Zapier probably expect extensions always, I think that is how they noticed
On our web one can download and listen the WAV file, but not preview it in the browser

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**